### ISSUE TITLE
Script sets can't be updated (added/removed)

### ISSUE THREAD

**Issue by knoajp** [JasonBarnabe/greasyfork#644]
I can't add a new script or remove an old script to any of the existing script set.
(I could add new script on "Included" list and "Calculated script list" was surely updated, but then I click the "Save" button, the results remained old list.)

Once you created a script set, you can't update it forever.
(you can just edit the descriptions of them)

**Comment by knoajp** [comment]
I found at least one script set which can be add a new script.
https://greasyfork.org/ja/scripts?set=361295

But the rest of my script set was not.
Even if the new set for a test.

**Comment by JasonBarnabe** [comment]
I tried a few actions, couldn't get an error. I did notice that saving the script set could fail silently, so I made it fail non-silently.

Can you try again and let me know exactly which set you're trying to edit, and what you're trying to do with it?

**Comment by knoajp** [comment]
This one.
https://greasyfork.org/ja/users/78927-knoa/sets/361295/edit

I added "YouTube Live Cpu Tamer" and seems to be successfully added on edit view,
https://greasyfork.org/ja/scripts/396532-youtube-live-cpu-tamer
and also get success message on saving, but it didn't be listed.

Does it take some time to appear?

**Comment by JasonBarnabe** [comment]
Shows up for me - https://greasyfork.org/en/scripts?set=361295

**Comment by knoajp** [comment]
Shows up for me too now.

I tried to add another script for a test and succeeded instantly.
But then I tried to remove the script and got failed.
I tried to refresh the page with no cache but failed.
I checked the script set page on another brwoser and successfully the script didn't listed.
Again I tried the former browser to refresh then got succeed.

It's quite complicated but I think it could be related on browser cache, server or database cache, login session, author of the script set, an access from another user, and so on.

**Comment by knoajp** [comment]
Shows up for me too now.

I tried to add another script for a test and succeeded instantly.
But then I tried to remove the script and got failed.
I tried to refresh the page with no cache but failed.
I checked the script set page on another brwoser and successfully the script didn't listed.
Again I tried the former browser to refresh then got faild.
I logged out on the former browser and got succeed!!
But again, I logged in and visit and refresh the page with no cache, I failed!! The script I've removed was listed again!!

It's quite complicated but I think it could be related on browser cache, server or database cache, login session, author of the script set, an access from another user, and so on.

**Comment by knoajp** [comment]
(I don't know it has any relations on the problem...)

On the script set page, the script data such as "Daily Installs" seems to keep rather old data.
https://greasyfork.org/en/scripts?set=361295
While identical data on the user page is well updated.
https://greasyfork.org/ja/users/78927-knoa

**Comment by JasonBarnabe** [comment]
This looks like a server caching issue. There were two problems:

1. updated_at was not updated when a script was removed from the set
2. The code wasn't working with Rails reusable cache keys so the cache would not expire on update anyway.

I've fixed these, so let me know if you see further issues.

The daily installs not matching is also a caching issue, but one that I'm fine with. It's not like I'm going to clear the cache every time someone installs.
